video/out/vo_drm.c

```diff
@@ -325,6 +345,158 @@ static int modeset_prepare_dev(struct vo *vo, int fd, int conn_id,
 
 
 
+static int vt_switcher_pipe[2];
+
+static void vt_switcher_sighandler(int sig)
+{
+    unsigned char event = sig == SIGUSR1 ? EVT_RELEASE : EVT_ACQUIRE;
+    write(vt_switcher_pipe[1], &event, sizeof(event));
+}
+
+static void vt_switcher_interrupt(struct vt_switcher *s)
+{
+    unsigned char event = EVT_INTERRUPT;
+    write(vt_switcher_pipe[1], &event, sizeof(event));
+}
+
+static int vt_switcher_init(struct vt_switcher *s, struct vo *vo)
+{
+    s->vo = vo;
+    if (pipe(vt_switcher_pipe)) {
+        MP_ERR(vo, "Creating pipe failed: %s", mp_strerror(errno));
+        return -1;
+    }
+
+    s->tty_fd = open("/dev/tty", O_RDWR);
```

Comment:
Generally, all created file descriptors should have the O_CLOEXEC flag set, or a forked process could access them. (There's a helper mp_make_cloexec_pipe(), which does this for pipes.)